lib/rubocop/node_pattern.rb

```diff
@@ -384,10 +384,13 @@ def compile_param(cur_node, number, seq_head)
       end
 
       def compile_args(tokens)
-        args = []
-        args << compile_arg(tokens.shift) until tokens.first == ')'
-        tokens.shift # drop the )
-        args
+        index = tokens.find_index { |token| token == ')' }
+
+        tokens.slice!(0..index).each_with_object([]) do |token, args|
```

Comment:
Is it safe to mutate the passed in array here?

Reply:
👋 , if you are talking about the `tokens` array then I believe it should be safe to mutate it, well at least the previous implementation was already mutating it